components/images/web3-image.tsx

```diff
@@ -0,0 +1,25 @@
+"use client";
+
+import React from "react";
+import Image, { ImageProps } from "next/image";
+import { web3ImgLoader } from "@/lib/web3-img-loader";
+
+const Web3Image = React.forwardRef<HTMLImageElement, ImageProps>(
+  ({ alt, src, ...props }, ref) => {
+    const isWeb3 = typeof src === "string" && src.startsWith("ipfs://");
+
+    return (
+      <Image
+        ref={ref}
+        src={src}
+        alt={alt}
+        loader={isWeb3 ? web3ImgLoader : undefined}
+        {...props}
+      />
+    );
+  },
+);
+
+Web3Image.displayName = "Web3Image";
+
+export default Web3Image;
```

Comment:
I try to avoid `export default` since having a named exports greatly helps import tooling (auto importer and symbol renaming namely)